fix(tags): Fix autocomplete to select the best match on Enter

hexed0ut opened the pull request:
Resolves #33747


> Explain the **details** for making this change. What existing problem does the pull request solve?

This update improves the tag autocompletion functionality by altering how the Enter key event is handled. Previously, pressing Enter simply added the current tag-input field as-is. With this fix, pressing Enter now triggers an event which updates the tag-input field with the best-matched suggestion from the available tags. After this field change, an `input-selected` event is fired, invoking the existing `select_tag` method to complete the tag addition.

> Screenshots/GIFs

https://github.com/user-attachments/assets/f10667dd-703a-4d9d-8594-6b3e6ac83737

hexed0ut committed:
fix(tags): Fix autocomplete to automatically select the best match

Resolves #33747

hexed0ut committed:
chore(tags): Add semi-colon

iamejaaz reviewed (changes requested):
Looks good, but there seems to be an issue: when I press the Enter key, it selects the matching tag, but the dropdown doesn’t close. As a result, I can’t see if the tag is selected. Could you please make the dropdown close on selection for better UX?

hexed0ut committed:
fix(tags): Close dropdown on tag selection

Context - https://github.com/frappe/frappe/pull/34625#pullrequestreview-3521774645

hexed0ut commented:
@iamejaaz Fixed. Thanks!

iamejaaz commented:
> @iamejaaz Fixed. Thanks!

@hexed0ut Thanks for making the changes. Could you please update the video in the description?

hexed0ut commented:
@iamejaaz Done!

hexed0ut commented:
Hey @iamejaaz, is there any other change you would like me to make?

iamejaaz commented:
@Mergifyio update

iamejaaz committed:
Merge branch 'develop' into fix-tags-autocomplete

iamejaaz commented:
> Hey @iamejaaz, is there any other change you would like me to make?

Apologies for the delay 😅  , and thank you for the reminder 😃 .

hexed0ut commented:
@iamejaaz No worries, thanks!
